db/crud.go

```diff
@@ -1476,63 +1470,50 @@ func (db *DatabaseCollectionWithUser) storeOldBodyInRevTreeAndUpdateCurrent(ctx
 	}
 }
 
-func (db *DatabaseCollectionWithUser) prepareSyncFn(doc *Document, newDoc *Document) (mutableBody Body, metaMap map[string]interface{}, newRevID string, err error) {
+func (db *DatabaseCollectionWithUser) prepareSyncFn(doc *Document, newDoc *Document) (revBody []byte, metaMap channels.MetaMap, newRevID string, err error) {
 	// Marshal raw user xattrs for use in Sync Fn. If this fails we can bail out so we should do early as possible.
 	metaMap, err = doc.GetMetaMap(db.userXattrKey())
 	if err != nil {
 		return
 	}
 
-	mutableBody, err = newDoc.GetDeepMutableBody()
+	revBody, err = newDoc.BodyBytes()
 	if err != nil {
 		return
 	}
 
-	err = validateNewBody(mutableBody)
+	err = validateNewBody(newDoc.Body())
 	if err != nil {
 		return
 	}
 
 	newRevID = newDoc.RevID
-
-	mutableBody[BodyId] = doc.ID
-	mutableBody[BodyRev] = newRevID
-	if newDoc.Deleted {
-		mutableBody[BodyDeleted] = true
-	}
-
 	return
 }
 
 // Run the sync function on the given document and body. Need to inject the document ID and rev ID temporarily to run
 // the sync function.
-func (db *DatabaseCollectionWithUser) runSyncFn(ctx context.Context, doc *Document, body Body, metaMap map[string]interface{}, newRevId string) (*uint32, string, base.Set, channels.AccessMap, channels.AccessMap, error) {
-	channelSet, access, roles, syncExpiry, oldBody, err := db.getChannelsAndAccess(ctx, doc, body, metaMap, newRevId)
+func (db *DatabaseCollectionWithUser) runSyncFn(ctx context.Context, doc *Document, body []byte, metaMap channels.MetaMap, newRevId string, deleted bool) (*uint32, string, base.Set, channels.AccessMap, channels.AccessMap, error) {
```

Comment:
why is it passing body separately helpful? Does it make sense to expose `Document._rawBody` here as a more clear method, or is there a mutability problem?